[FSSS-199] Implement/style Search results page

heitorado opened the pull request:
## What's the purpose of this pull request?
This PR styles the search page results. Below are the images for reference.

- Desktop
<img width="975" alt="Screen Shot 2022-03-16 at 17 36 03" src="https://user-images.githubusercontent.com/24229855/158686169-134d1b23-bcc5-477e-80ad-495e2ee42378.png">
<img width="974" alt="Screen Shot 2022-03-16 at 17 36 30" src="https://user-images.githubusercontent.com/24229855/158686219-db260be0-41b0-427c-ba70-5a427bcf6295.png">

- Mobile
<img width="291" alt="Screen Shot 2022-03-16 at 17 37 24" src="https://user-images.githubusercontent.com/24229855/158686339-0bfee18c-9dcd-481d-b056-9f1cc3017cf7.png">
<img width="292" alt="Screen Shot 2022-03-16 at 17 37 41" src="https://user-images.githubusercontent.com/24229855/158686389-bb7c1b69-5877-42df-90c5-0fc6788e7f2f.png">

## How does it work?
Nothing unusual, just some extra props being passed and changes/additions in HTML/CSS here and there.

## How to test it?
- `yarn test`, verify that all tests still pass
- Run the development server
  - Search something that exists in the store (I usually use the word `tasty`)
    - Verify that the search page results show up and it is styled like the images above, both on mobile and desktop.
  - Search something that does not exist in the store (I usually use the phrase `brastemp washing machine`) 
    - Verify that the empty gallery (no results found) page renders and it is styled like the images above, both on mobile and desktop

## References
Thanks to @danzanzini for all the help

## Checklist
- [x] CHANGELOG entry added

Commit:
Enhance ProductTitle's docs

Commit:
Add 'searchTerm' parameter to Product/EmptyGallery

Commit:
Style EmptyGallery

Commit:
Style ProductGallery

Commit:
Finish styling EmptyGallery

Co-authored-by: Daniel Zanzini <danzanzini@users.noreply.github.com>

Commit:
Enhance ProductGallery HTML semantics

Commit:
Remove possibly misplaced docs

Commit:
Add and style Breadcrumb to search results

Commit:
Revert Empty State/Gallery to last implementation

Just add padding and extra breakpoint rule

Commit:
Update CHANGELOG

Commit:
Empty commit to trigger CI

eduardoformiga commented on src/components/sections/ProductGallery/product-gallery.scss:
```diff
@@ -4,12 +4,25 @@
+  margin-bottom: var(--space-3);
+
+  h1 {
+    font-size: var(--text-size-5);
```
I think on mobile (mobile first) we should use the `--text-size-4` (25px)

```suggestion
    font-size: var(--text-size-4);

    @include media(">=tablet") { font-size: var(--text-size-5); }
```

eduardoformiga reviewed (approved):
Good Job! 👏 

I left some minor comments.

heitorado commented on src/pages/s.tsx:
```diff
@@ -54,13 +55,14 @@ function Page(props: Props) {
         }}
       />
 
-      {/*
-        Sections: Components imported from '../components/sections' only.
-        Do not import or render components from any other folder in here.
-      */}
```
I agree, I just do not think this was the right place for it. Maybe we should leave this only inside the `Section` component?

filipewl replied:
I guess this comment didn't stop us from not following it when we were making our changes. But we should've paid more attention.
😆 

I like having more details about why this is needed on the documentation somewhere.

And if we're going to remove it, we should remove them altogether (there are 2 other occurrences in the Home and PDP pages).

heitorado replied:
This was making sense to me some commits ago, where the `Section` component was being used explicitly in that file as a wrapper for other components, so the warning was coherent: It should warn us that not everything could be put inside the `Section` component, just the ones inside `components/sections` (I think).

However, some commits later I noticed that it vanished from that file (it was moved), and now the `Section` component resides inside each component it wraps (look at `Breadcrumb.tsx` for an example) - it is not being called "from outside" anymore. But the comment remains.

heitorado replied:
@filipewl more about sections (and the `Section` component) here: https://github.com/vtex-sites/base.store/pull/371

TL;DR:
> The whole idea of sections is to split app logic into replaceable/reusable strips of screen content so that CMS users can easily replace/rearrange the desired component on the screen. (...)
> This means that the Section map has to render a self contained section. 
> (...) so CMS users can correctly change and re-order the sections

heitorado replied:
I'll rewrite and bring back the comments, now that their purpose is clearer.

Commit:
Update src/components/sections/ProductGallery/product-gallery.scss

Co-authored-by: Eduardo Formiga <eduardo.formiga@gmail.com>

filipewl commented on src/components/sections/ProductGallery/product-gallery.scss:
```diff
@@ -4,12 +4,27 @@
   --product-listing-row-height: var(--space-6);
 
   height: 100%;
+  padding-top: 0;
```
This seems to be needed for the search page, but the existing padding is needed for PLP. 
😕 

https://user-images.githubusercontent.com/381395/158908807-66f4f054-4c76-4670-9730-c5a05529cb25.mp4

![CleanShot 2022-03-17 at 20 09 31](https://user-images.githubusercontent.com/381395/158908842-470ccf14-f87b-458f-b71e-dd23a270374b.png)

Now how to solve this only for the search page... 
🤔

heitorado replied:
After aligning with the designers we changed the design a bit to accomodate the title gracefully, not interfering with the PLP style.

heitorado replied:
After aligning with the designers we changed the design a bit to accommodate the title gracefully, not interfering with the PLP style.

filipewl reviewed (commented):
Looking very good! 👏

filipewl commented:
I wonder if we should include the search page to be run on LH to make sure it's consistent with the rest?

filipewl commented:
Does this task also encompass this option for clearing the search term?

![CleanShot 2022-03-17 at 20 25 19](https://user-images.githubusercontent.com/381395/158909942-22ec9e01-5ee3-4ef5-8d20-3193849812e5.png)

heitorado commented:
> Does this task also encompass this option for clearing the search term?
> 
I've aligned with the designers and we won't be doing that, since it will be removed soon on the next version.

Commit:
Fix padding on search page breaking plp
- Change the design a bit to make it work with both plp and search page
- Change media query that was not mobile-first

Commit:
Add back docs about sections with improvements

filipewl reviewed (approved):
Looks good to me!

Commit:
Empty commit to trigger CI

Commit:
Empty commit to trigger CI
